addOns/callhome/src/main/java/org/zaproxy/addon/callhome/ExtensionCallHome.java

```diff
@@ -0,0 +1,235 @@
+        JSONObject json = new JSONObject();
+        json.put("zapVersion", Constant.PROGRAM_VERSION);
+        json.put("os", getOS().toString());
+        json.put(
+                "osVersion",
+                System.getProperty("os.name") + " : " + System.getProperty("os.version"));
+        json.put("javaVersion", System.getProperty("java.version"));
+        json.put("zapType", ZAP.getProcessType().name());
+        json.put("container", Constant.isInContainer() ? Constant.getContainerName() : "");
+        return json;
+    }
+
+    public ZapXmlConfiguration getCheckForUpdatesData()
+            throws IOException, ConfigurationException, InvalidCfuUrlException {
+        LOGGER.debug("Getting latest version info from {}", ZAP_CFU_SERVICE);
+
+        HttpMessage msg = new HttpMessage(new URI(ZAP_CFU_SERVICE, true));
+        msg.getRequestHeader().setMethod(HttpRequestHeader.POST);
+        msg.getRequestHeader().setHeader(HttpHeader.CONTENT_TYPE, HttpHeader.JSON_CONTENT_TYPE);
+        msg.getRequestBody().setBody(getMandatoryRequestData().toString());
+        msg.getRequestHeader().setContentLength(msg.getRequestBody().length());
+
+        getHttpSender().sendAndReceive(msg, true);
+        if (msg.getResponseHeader().getStatusCode() != HttpStatusCode.OK) {
+            throw new IOException(
+                    "Expected '200 OK' but got '"
+                            + msg.getResponseHeader().getStatusCode()
+                            + " "
+                            + msg.getResponseHeader().getReasonPhrase()
+                            + "'");
+        }
+        if (!msg.getRequestHeader().isSecure()) {
+            // Only access the cfu page over https
+            throw new InvalidCfuUrlException(msg.getRequestHeader().getURI().toString());
+        }
+
+        ZapXmlConfiguration config = new ZapXmlConfiguration();
+        config.setDelimiterParsingDisabled(true);
+        config.load(new StringReader(msg.getResponseBody().toString()));
+
+        return config;
+    }
+
+    /** Get the latest ZAP Check for Updates data. This is used by the core from 2.12.0. */
+    @Override
+    public ZapXmlConfiguration get() {
+        try {
+            return getCheckForUpdatesData();
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+        return null;
+    }
+
+    public static OS getOS() {
+        if (os == null) {
+            if (Constant.isWindows()) {
+                os = OS.WINDOWS;
+            } else if (Constant.isKali()) {
+                os = OS.KALI;
+            } else if (isBackBox()) {
+                os = OS.BACK_BOX;
+            } else if (Constant.isLinux()) {
+                os = OS.LINUX;
+            } else if (Constant.isMacOsX()) {
+                os = OS.MAC_OS;
+            } else {
+                os = OS.UNKNOWN;
+            }
+        }
+        return os;
+    }
+
+    // XXX: Use Constant.isBackBox() from 2.12.0
```

Comment:
I believe this comment can go? It's being dropped from core in the other PR: https://github.com/zaproxy/zaproxy/pull/6932/

Reply:
No, it should still be there :/